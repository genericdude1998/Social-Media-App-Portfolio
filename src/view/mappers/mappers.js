import { doCloseComments, doGetPostsThunk, doGetUserThunk, doOpenComments, doSendCommentThunk, doSendPostThunk, doSendUsernameAndPasswordThunk, doSetCommentContent, doSetPassword, doSetPostContent, doSetUsername, doToggleTheme } from '../../model/actionCreators/actionCreators';
import { loginStateSelector, feedStateSelector, userInfoStateSelector, NPCStateSelector, themeStateSelector, NCCStateSelector } from '../../model/selectors/selectors';

export function mapLoginDispatchToProps(dispatch){
    return {
        onChangeUsername: (e) => dispatch(doSetUsername(e.target.value)),
        onChangePassword: (e) => dispatch(doSetPassword(e.target.value)),
        onSubmitLogin: (username, password) => (event) => dispatch(doSendUsernameAndPasswordThunk(username,password, event)),
    };
}
export function mapLoginStateToProps(state){
    return loginStateSelector(state);
}

export function mapFeedDispatchToProps(dispatch){
    return {
        onFetchPosts: () => dispatch(doGetPostsThunk()),
    };
}
export function mapFeedStateToProps(state){
    return feedStateSelector(state);
}

export function mapUserDispatchToProps(dispatch){
    return {
        onFetchUser: (id) => dispatch(doGetUserThunk(id)),
    };
}
export function mapUserStateToProps(state){
    return userInfoStateSelector(state);
}

export function mapNPCDispatchToProps(dispatch){
    return {
        onSetPostContent: (e) => dispatch(doSetPostContent(e.target.value)),
        onSendNewPost: (content, navigate) => (event) => dispatch(doSendPostThunk(content, event, navigate)),
    };
}
export function mapNPCStateToProps(state){
    return NPCStateSelector(state);
}

export function mapThemeDispatchToProps(dispatch){
    return {
        onToggleTheme: () => dispatch(doToggleTheme()),
    };
}
export function mapThemeStateToProps(state){
    return themeStateSelector(state);
}

export function mapCommentsDispatchToProps(dispatch){
    return {
        onOpenComments: (postId) => dispatch(doOpenComments(postId)),
        onCloseComments: (postId) => dispatch(doCloseComments(postId)),
    };
}

export function mapNCCDispatchToProps(dispatch){
    return {
        onSetCommentContent: (e) => dispatch(doSetCommentContent(e.target.value)),
        onSendNewComment: (content, navigate, postId) => (event) => dispatch(doSendCommentThunk(content, event, navigate, postId)),
    };
}
export function mapNCCStateToProps(state){
    return NCCStateSelector(state);
}