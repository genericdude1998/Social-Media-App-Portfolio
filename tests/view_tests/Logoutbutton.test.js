import React from 'react';
import { shallow } from 'enzyme';
import { connect } from 'react-redux';
import ConnectedLogoutButton, { Logoutbutton } from '../../src/view/Feed/LogoutButton';
import { mapLogoutDispatchToProps } from '../../src/view/mappers/mappers';

jest.mock('react-redux', () => {
    return {
        connect: jest.fn(() => {
            return jest.fn().mockImplementation(ReactComponent => {
                return ReactComponent
            });
        }) ,
        Provider: ({ children }) => children,
    }
});

describe('ConnectedLogOutButton', () => {
    it('should call connect with expected params', () => {
        expect(connect).toHaveBeenCalledWith(undefined, mapLogoutDispatchToProps);
        expect(ConnectedLogoutButton).toBe(Logoutbutton);
    });
    it('should matchSnapshot', () => {
        const wrapper = shallow(<Logoutbutton onLogoutUser={jest.fn()} />);
        expect(wrapper).toMatchSnapshot();
    });
});
