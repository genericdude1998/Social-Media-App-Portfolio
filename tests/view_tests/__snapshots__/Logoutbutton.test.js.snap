// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConnectedLogOutButton should matchSnapshot 1`] = `
<NavLink
  onClick={[MockFunction]}
  onLogoutUser={[MockFunction]}
/>
`;
